### DISCUSSION TITLE
Question regarding clientExtensions

### DISCUSSION THREAD

**Discussion by rafbgarcia** [prisma/prisma#16693] [Q&A] [answered]
Hello!

Is there a way to override the transaction behavior for all built-in functions that use transactions?

The reason I ask is to implement what's discussed on this issue: https://github.com/prisma/prisma/issues/11183

I've tried the following code but I can't manage to get it working: https://github.com/prisma/prisma/issues/11183#issuecomment-1341955066

I appreciate any insights. Thank you.

**Comment by nurul3101** [comment, marked as answer]
Hello @rafbgarcia 👋 

How are you invoking the `$transaction` function from the Extended Prisma Client?
I tried executing this file and it seems to work as expected for me

```typescript
import { Prisma, PrismaClient } from '@prisma/client';

const prisma = new PrismaClient();

const xprisma = prisma.$extends({
  client: {
    $transaction(callback: any) {
      console.log('callback', callback);
      return prisma.$transaction(async (tx: any) => {
        const res = await prisma.$executeRawUnsafe(
          "SELECT table_name FROM information_schema.tables WHERE table_schema='public' AND table_type='BASE TABLE'"
        );
        console.log('res', res);
      });
    },
  },
});

async function main() {
  const tx = await xprisma.$transaction('test');
}

main()
  .catch((e) => {
    throw e;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

```

Here's the response:
<img width="522" alt="Screenshot 2022-12-09 at 8 22 21 PM" src="https://user-images.githubusercontent.com/13049676/206729034-c8ad8c0b-c8b6-4712-9a44-a5062819ff68.png">

Please let me know if I misunderstood your question.

**Reply by rafbgarcia** [reply]
Hi @nurul3101 thanks for your response!

What I'd like to do is for that transaction logic to be applied anywhere Prisma uses transactions, such as `prisma.model.create`.

Is that possible?

**Reply by nurul3101** [reply]
I don't think this is possible yet, the default transaction behaviour would not be overwritten by extending Prisma Client.

**Reply by rafbgarcia** [reply]
Thanks @nurul3101 !

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
